apps/dot_ext/loggers.py

```diff
@@ -0,0 +1,199 @@
+        except Application.DoesNotExist:
+            pass
+
+
+def update_session_auth_flow_trace(request=None, auth_uuid=None, state=None, code=None):
+    '''
+    Function for updating auth flow related items in the
+    AuthFlowUuid tracking object and session.
+
+    Summary of cases based on arguments passed in:
+
+    - auth_uuid and code = Update previously created AuthFlowUuid instance with code.
+    - code =  Get session values from AuthFlowUuid via code.
+    - auth_uuid = Get session values from AuthFlowUuid.
+    - state = Update or set state in session.
+    '''
+    # Set to None for Unit tests to work.
+    auth_flow_uuid = None
+
+    if auth_uuid and code:
+        # Get and update previously created AuthFlowUuid instance with code.
+        try:
+            auth_flow_uuid = AuthFlowUuid.objects.get(auth_uuid=auth_uuid)
+            # Set code.
+            auth_flow_uuid.code = code
+            auth_flow_uuid.save()
+        except AuthFlowUuid.DoesNotExist:
+            # Create AuthFlowUuid instance, if it doesn't exist.
+            if auth_uuid:
+                try:
+                    AuthFlowUuid.objects.create(auth_uuid=auth_uuid, code=code, state=None)
+                except IntegrityError:
+                    pass
+
+    elif code:
+        # Get session values from AuthFlowUuid via code.
+        try:
+            # Get previously created AuthFlowUuid with code.
+            auth_flow_uuid = AuthFlowUuid.objects.get(code=code)
+            set_session_values_from_auth_flow_uuid(request, auth_flow_uuid)
+
+            # Delete the no longer needed instance
+            auth_flow_uuid.delete()
+        except AuthFlowUuid.DoesNotExist:
+            pass
+
+    elif auth_uuid:
+        # Get session values from AuthFlowUuid.
+        request.session['auth_uuid'] = str(auth_uuid)
+
+        try:
+            auth_flow_uuid = AuthFlowUuid.objects.get(auth_uuid=auth_uuid)
+            set_session_values_from_auth_flow_uuid(request, auth_flow_uuid)
+        except AuthFlowUuid.DoesNotExist:
+            pass
+
+    elif state:
+        # Update AuthFlowUuid instance to pass along auth_uuid using state.
+        auth_uuid = request.session.get('auth_uuid', None)
+
+        if auth_uuid:
+            # Store state in AuthFlowUuid if auth_uuid exists.
+            try:
+                auth_flow_uuid = AuthFlowUuid.objects.get(auth_uuid=auth_uuid)
+                auth_flow_uuid.state = state
+                auth_flow_uuid.save()
+            except AuthFlowUuid.DoesNotExist:
+                auth_flow_uuid = None
+
+        if not auth_flow_uuid:
+            # Retreive auth flow session values using previous state in AuthFlowUuid.
+            try:
+                auth_flow_uuid = AuthFlowUuid.objects.get(state=state)
+                set_session_values_from_auth_flow_uuid(request, auth_flow_uuid)
+            except AuthFlowUuid.DoesNotExist:
+                pass
```

Comment:
I think this is another case where this try statement can live directly within the proceeding try statement.